### PR TITLE
IDEX-3267 Return any-any permissions for all VFS entries

### PR DESCRIPTION
**Pull request by skabashnyuk**
In 4.0 we are not relying on vfs permission so we can simply return any-any permission
for all users. We need to return something because UI is still relying on this information.

### CODE REVIEW

File: che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java [hunk 1]
```diff
@@ -296,6 +250,10 @@ protected AccessControlList loadValue(Path key) {
 
     private final VirtualFileSystemUserContext userContext;
 
+
+
+    private final AccessControlList DEFAULT_ACL = new AccessControlList();
```
Comment (evoevodin) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: It is not a constant, so it should not be defined in `CONSTANT_CASE`

File: che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java [hunk 2]
```diff
@@ -323,11 +278,27 @@ protected AccessControlList loadValue(Path key) {
...
+
+        acl.add(newDto(AccessControlEntry.class)
+                        .withPrincipal(newDto(Principal.class)
+                                               .withName("temp_user")
+                                               .withType(Principal.Type.GROUP))
+                        .withPermissions(singletonList("all")));
+        DEFAULT_ACL.update(acl, true);
+
+
+
```
Comment (evoevodin) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: Why there are 4 empty lines?

File: che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java [hunk 3]
```diff
@@ -297,6 +249,11 @@ protected AccessControlList loadValue(Path key) {
     private final VirtualFileSystemUserContext userContext;
 
     /**
+     * ACL that returned for any fs entry.
+     */
+    private final AccessControlList defaultAcl = new AccessControlList();
```
Comment (benoitf) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: I would say that it needs to be static as well as it's immutable and not defined per instance

Comment (skabashnyuk) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: As we expect in near future to remove this code at all I would say such complexity is not necessary. The purpose of this change is to make things working until we remove VFS at all.

Comment (benoitf) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: so it needs to introduce @deprecated stuff as well

Comment (skabashnyuk) [inline che-core-vfs-impl/src/main/java/org/eclipse/che/vfs/impl/fs/FSMountPoint.java]: Module marked as deprecated

**Comment by benoitf** [comment]
+1

**Pull request merged**